packages/cadl-ranch-specs/http/lro/lro-core/main.cadl

```diff
@@ -0,0 +1,204 @@
+import "@azure-tools/cadl-azure-core";
+import "@azure-tools/cadl-ranch-expect";
+import "@cadl-lang/rest";
+
+using Azure.Core;
+using global.Azure.Core.Traits;
+using Cadl.Rest;
+using Cadl.Http;
+
+#suppress "@azure-tools/cadl-azure-core/casing-style" "For spec"
+@doc("Illustrates bodies templated with Azure Core with long-running operation")
+@scenarioService("/azure/lro/core")
+namespace Azure.Lro.Core;
+
+interface ResourceOperations
+  extends global.Azure.Core.ResourceOperations<NoConditionalRequests & NoRepeatableRequests & NoClientRequestId> {}
+
+@resource("users")
+@doc("Details about a user.")
+model User {
+  @key
+  @visibility("read")
+  @doc("The name of user.")
+  name: string;
+
+  @doc("The role of user")
+  role: string;
+}
+
+@doc("The parameters for exporting a user.")
+model UserExportParams {
+  @query
+  @doc("The format of the data.")
+  format: string;
+}
+
+@doc("The exported user data.")
+model ExportedUser {
+  @doc("The name of user.")
+  name: string;
+
+  @doc("The exported URI.")
+  resourceUri: string;
+}
+
+@scenario
+@doc("Creates or repalces a User")
+@summary("Adds a user or repalces a user's fields.")
+@scenarioDoc("""
+Should only generate one model named User.
+
+Expected verb: PUT
+Expected path parameter: name=madge
+
+Expected request body:
+```json
+{
+  "role": "contributor"
+}
+```
+
+Expected status code: 201
+Expected response header: operation-location={endpoint}/users/madge/operations/operation1
+Expected response body:
+```json
+{
+  "name": "madge"
+  "role": "contributor"
+}
+```
+
+Expected verb: GET
+Expected URL: {endpoint}/users/madge/operations/operation1
+
+Expected status code: 200
+Expected response body:
+```json
+{
+  "id": "operation1",
+  "status": "InProgress"
+}
+```
+
+Expected verb: GET
+Expected URL: {endpoint}/users/madge/operations/operation1
+
+Expected status code: 200
+Expected response body:
+```json
+{
+  "id": "operation1",
+  "status": "Succeeded"
+}
+```
+
+Expected verb: GET
+Expected URL: {endpoint}/users/madge
+
+Expected status code: 200
+Expected response body:
+```json
+{
+  "name": "madge"
+  "role": "contributor"
+}
+```
```

Comment:
In old "PUT to create" or "POST to create" scenario (the "operation resource" pattern), the last "status=Succeeded" response of status monitor would contains a "resourceLocation" property to point to the URL of the resource created.

This won't be there in this new design.

In Guideline, the last GET on resource URL (`{endpoint}/users/madge`) is not required. But I am not sure if SDK would still want to do this, to get the final result (in case any property of the resource changes from the first 201 from PUT and this 200 from GET).